apis/nucleus/src/components/listbox/components/ListBoxSearch.jsx

```diff
@@ -7,46 +7,67 @@ import InstanceContext from '../../../contexts/InstanceContext';
 
 const TREE_PATH = '/qListObjectDef';
 
-export default function ListBoxSearch({ model, keyboard, dense = false, visible = true }) {
+export default function ListBoxSearch({ selections, model, keyboard, dense = false, visible = true }) {
   const { translator } = useContext(InstanceContext);
   const [value, setValue] = useState('');
-  const [isSearching, setIsSearching] = useState(false);
+  const [searchApplied, setSearchApplied] = useState(false);
   const theme = useTheme();
 
   const abortSearch = async () => {
-    await model.abortListObjectSearch(TREE_PATH);
-    setIsSearching(false);
+    try {
+      await model.abortListObjectSearch(TREE_PATH);
+    } finally {
+      setValue('');
+      setSearchApplied(false);
+    }
   };
 
-  useEffect(() => {
-    if (!visible && isSearching) {
-      abortSearch();
+  const handleDeactivated = () => {
+    if (!searchApplied) {
+      return;
     }
-  }, [visible]);
+    abortSearch();
+  };
+
+  useEffect(() => {
+    selections.on('deactivated', handleDeactivated);
+    return () => {
+      selections.removeListener && selections.removeListener('deactivated', handleDeactivated);
+    };
+  }, []);
 
-  const onChange = (e) => {
+  const onChange = async (e) => {
     setValue(e.target.value);
-    if (e.target.value.length === 0) {
-      abortSearch();
-    } else {
-      model.searchListObjectFor(TREE_PATH, e.target.value);
-      setIsSearching(true);
+    if (e.target.value.length) {
+      setSearchApplied(true);
+      return model.searchListObjectFor(TREE_PATH, e.target.value);
     }
+    return undefined;
   };
+
+  const handleFocus = () => {
+    if (!selections.isModal(model)) {
+      selections.begin(['/qListObjectDef']).then(() => {
+        selections.goModal && selections.goModal('/qListObjectDef');
```

Comment:
why?
it looks like selections.begin already are doing this

Reply:
Yes, it can be removed.